src/charm.py

```diff
@@ -205,6 +187,74 @@ def _check_kf_profiles(self, interfaces):
 
         return kf_profiles
 
+    def _create_resources(self, resource_type: List[str] = None) -> None:
+        """Creates the resources for the charm."""
+        if resource_type is None:
+            resource_type = self._resource_files.keys()
+        for resource_type in resource_type:
+            resource_file = self._resource_files[resource_type]
+            manifest = self.env.get_template(resource_file).render(self._context)
+            for obj in codecs.load_all_yaml(manifest):
+                try:
+                    self.lightkube_client.create(obj)
+                except ApiError as e:
+                    if e.status.code == 409:
+                        self.logger.info("replacing resource: %s.", str(obj.to_dict()))
+                        self.logger.debug(f"manifest is {manifest}")
+                        try:
+                            self.lightkube_client.patch(
+                                type(obj),
+                                obj.metadata.name,
+                                obj,
+                                patch_type=PatchType.MERGE,
+                            )
+                        except ApiError as e:
+                            if e.status.code == 409:
+                                self.logger.info(
+                                    "Unable to replace resource: %s. Skipping.",
+                                    str(obj.to_dict()),
+                                )
+                            else:
+                                raise e
+                    else:
+                        self.logger.debug(
+                            "failed to create resource: %s.", str(obj.to_dict())
+                        )
+                        raise e
+
+    def main(self, event) -> None:
+        """Main entry point for the Charm."""
+        try:
+            self._check_container_connection()
+            self._check_model_name()
+            self._check_leader()
+            interfaces = self._get_interfaces()
+            kf_profiles = self._check_kf_profiles(interfaces)
+        except CheckFailed as e:
+            self.model.unit.status = e.status
+            return
+        try:
+            self.unit.status = MaintenanceStatus("Creating k8s resources")
+            try:
+                self._create_resources(["auths"])
+                current_configmap = self.lightkube_client.get(
+                    ConfigMap, name=self._context["configmap_name"]
+                )
+            except Exception as e:
+                self.logger.info(f"ConfigMap not found: {e}. Creating one")
```

Comment:
As is, we get here for any error that comes up.  Should this catch only the errors raised by lightkube when an object is missing (I think it is something like a 404 error?  You'll need to check)

Reply:
Good catch